### PR TITLE
Show the new point on the nearest wall

### PR DESCRIPTION
**Pull request by SkelNeXus**

### CODE REVIEW

**Comment by SkelNeXus** [comment]
There is a bug on drag room, I need help to fix it @SchawnnDev

File: src/app/core/commands/wall-commands.ts [hunk 1]
```diff
@@ -137,76 +138,38 @@ export class EditLastWallWithPointCommand extends Command {
   }
 }
 
-export class FinaliseRoomCommand extends Command {
+export class DivideWallCommand extends Command {
+  private newWall: Wall | null = null;
```
Comment (SchawnnDev) [inline src/app/core/commands/wall-commands.ts:142]:
```suggestion
  private newWall?: Wall;
```

File: src/app/core/commands/wall-commands.ts [hunk 2]
```diff
@@ -137,76 +138,38 @@ export class EditLastWallWithPointCommand extends Command {
...
 
-  constructor() {
+  constructor(
+    private wall: Wall,
+    private clickablePoint : ClickablePoint | undefined = undefined,
```
Comment (SchawnnDev) [inline src/app/core/commands/wall-commands.ts:146]:
```suggestion
    private clickablePoint?: ClickablePoint,
```

File: src/app/core/commands/wall-commands.ts [hunk 3]
```diff
@@ -137,76 +138,38 @@ export class EditLastWallWithPointCommand extends Command {
...
-    this.board.currentRoom = undefined;
-    this.board.drawState = DrawState.None; // When a room is finalized, the user should not be able to draw walls
-    this.board.normalisePoints();
-  }
-
-  override undo(): void {
-    const room = this.board.rooms.pop();
-    if (!room) { // Should never happen, but just in case
-      return;
+    if( this.clickablePoint == undefined ){
```
Comment (SchawnnDev) [inline src/app/core/commands/wall-commands.ts:153]:
```suggestion
    if(!this.clickablePoint){
```

File: src/app/core/models/board.ts [hunk 4]
```diff
@@ -12,6 +12,7 @@ import {ClickablePoint} from "./clickable-point";
 
 export class Board implements Drawable {
   public rooms: Room[];
+  public tempoDrawingElement: Drawable[];
```
Comment (SchawnnDev) [inline src/app/core/models/board.ts:15]:
```suggestion
  public tempDrawableElements: Drawable[];
```

File: src/app/core/models/wall.ts [hunk 5]
```diff
@@ -277,4 +278,12 @@ export class Wall extends Clickable implements Cloneable<Wall> {
     // Add the projection at the beginning of the segment to obtain the coordinates of the projected point
     return new Point(this.p1.x + projection.x, this.p1.y + projection.y);
   }
+
+  /**
+   * Check if the point is in the wall
+   * @param point The point to check
+   */
+  isCorrectlyPrintOnWall (point : Point) {
```
Comment (SchawnnDev) [inline src/app/core/models/wall.ts:286]:
```suggestion
  isPointOnSegment(point: Point) {
```

**Review by SchawnnDev — CHANGES_REQUESTED** [review]

**Review by SchawnnDev — APPROVED** [review]

**Pull request merged**